File: java/leetcode/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    static ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode root = new ListNode(0)
        root.next = head;
        ListNode cur = root;
        ListNode runner = root;

        for (int i = 0; i < n; i++) {
            runner = runner.next;
        }

        while (runner.next != null) {
            cur = cur.next;
            runner = runner.next;
        }
        cur.next = cur.next.next;
        return root.next;
    }
}

File: python/leetcode/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head, n):
        root = ListNode(0)
        root.next = head
        cur = root
        runner = root

        # look up use of `times` type block since i is not used
        for i in range(n):
            runner = runner.next

        while runner.next is not None:
            cur = cur.next
            runner = runner.next

        cur.next = cur.next.next

        return root.next
